test/Microsoft.AspNetCore.Hosting.Tests/WebHostTests.cs

```diff
@@ -121,6 +123,31 @@ public void UsesNewConfigurationOverLegacyConfigForAddresses()
         }
 
         [Fact]
+        public void DoNotPreferHostingUrlsWhenNoAddressConfigured()
+        {
+            var host = CreateBuilder().UseServer(this).PreferHostingUrls().Build();
+            host.Start();
```

Comment:
Ok, it's mostly for consistency - I see other tests disposing it. Not sure all of them do though.